src/PrestaShopBundle/Form/Admin/Extension/DisablingToggleExtension.php

```diff
@@ -0,0 +1,96 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Extension;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class DisablingToggleExtension extends AbstractTypeExtension
+{
+    /**
+     * @var EventSubscriberInterface
+     */
+    private $disablingToggleListener;
+
+    public function __construct(
+        EventSubscriberInterface $disablingToggleListener
+    ) {
+        $this->disablingToggleListener = $disablingToggleListener;
+    }
+
+    public const FIELD_PREFIX = 'disabling_toggle_';
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getExtendedTypes(): iterable
+    {
+        return [
+            // To add the option on all form types
+            FormType::class,
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        if ($builder->getOption('disabling_toggle')) {
+            $builder->addEventSubscriber($this->disablingToggleListener);
+        }
+    }
+
+    /**
+     * @param FormView $view
+     * @param FormInterface $form
+     * @param array $options
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options): void
+    {
+        $view->vars = array_replace($view->vars, [
+            'disabling_toggle' => $options['disabling_toggle'],
+        ]);
+    }
+
+    /**
+     * @param OptionsResolver $resolver
+     */
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver
+            ->setDefaults(['disabling_toggle' => false])
+            ->setAllowedTypes('disabling_toggle', ['array', 'bool'])
```

Comment:
As I suggested in another comment if you need a more complex format for configuration you should add a second option field to store the configuration